### DISCUSSION TITLE
Creating a factory resources dashboard

### DISCUSSION THREAD

**Discussion by cvetkovski98** [united-manufacturing-hub/united-manufacturing-hub#983] [General]
Hello guys,

I am trying to create a dashboard-like application that manages factory products, assets, and orders (general CRUD operations). For that reason, I was looking into consuming a REST API from the `factoryinsight` service but all I could find was the one designed for Grafana that returns a higher-level abstraction of data table-like entities. In my use case, a more traditional REST API where the data is represented in an array of JSON objects or similar with the added functionalities of creating, deleting, and updating entities would provide a greater value.

Is there any way to interface with the UMH system to query and update database data other than the aforementioned API? Do you have any other suggestions or ideas on how I should approach interfacing with the UMH system?

 Thanks for your time.

**Comment by JeremyTheocharis** [comment]
Hi @cvetkovski98,

sorry for the late response. Your message somehow went under.

To update the database we've been working on factoryinput. The general idea is never to change something in the database directly, but to always work via the message broker (to ensure that the update of a new shift, etc. is send to all other connected devices). There are some challenges in terms of security if you want to send data from Grafana to a message broker, this is what grafana-proxy and factoryinput are for.

If you want you can take a look there and if you have any questions or if stuff is not working, you can contact us! Looping in Ferdinand as well as he wrote the factoryinput integration. @Scarjit 

Hope this message does not come too late :)
